src/Components/Components/src/PersistenceContext.cs

```diff
@@ -0,0 +1,6 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components;
+
+internal readonly record struct PersistenceContext(IDictionary<string, byte[]> State);
```

Comment:
Maybe rename to `PersistentComponentStateStore`? Also does it have do operate on `IDictionary` or could it be `Dictionary`?

Reply:
I've reverted all this stuff to the old implementation, it's not needed.